### PR TITLE
Add auto cutscene skip option

### PR DESCRIPTION
**Pull request by Zedb0T**
Disabled by default but if enabled automatically skip cutscenes when possible.
Setting is shared between jak 1 and jak 2

Tested flutflut cutscene
Tested cell buying cutscenes
Tested final boss
Tested rats/fisherman
Tested failing rats/fisherman
Tested misty boat
Tested jak 2 onin game
Tested jak 2 mission fail
Tested jak 2 final boss

https://github.com/open-goal/jak-project/assets/89345505/245c6cde-7015-48f9-9ac7-e962ce0c4db8

### CODE REVIEW

File: goal_src/jak1/engine/common-obs/process-taskable.gc [hunk 1]
```diff
@@ -431,8 +431,11 @@
            (the-as spool-anim arg1)
            arg0
            (-> self blend-on-exit)
-           (lambda ((arg0 process-taskable)) (= (get-response (-> arg0 query)) 'no))
+           (if (and (-> *pc-settings* autoskip-movies?) (-> *pc-settings* skip-movies?))
+            (lambda ((arg0 process-taskable)) #t)
+            (lambda ((arg0 process-taskable)) (= (get-response (-> arg0 query)) 'no))
            )
+          )
```
Comment (ManDude) [inline goal_src/jak1/engine/common-obs/process-taskable.gc:438]: wrong indenting

**Review by ManDude — CHANGES_REQUESTED** [review]

**Comment by Zedb0T** [comment]
If we are resetting the settings anyways, should we also look to merge #3275  at the same time?

**Comment by Zedb0T** [comment]
Merge master and afaik this should be ready for a review.

**Comment by xTVaser** [comment]
Cleaning up stale PRs - this PR has sat for a while with feedback and no movement, re-open if/when work resumes.

**Comment by Zedb0T** [comment]
I changed the menu to the updated Jak 2 format sorry wasn't aware that the standard for that changed while this PR was waiting/sitting. Outside of that I really wasn't aware of anything that need ed to be done with this PR? If there was something please feel to advise. 

There was a Lint failure on the original PR but it was just a very very intentional change to the text of a og::preserve and the linter was mistaken in thinking that I was removing it.

**Comment by Zedb0T** [comment]
Discovered a bug on this, Just documenting it real quick it wasnt observed before the work on 2/24 so it shouldn't take too long to fix, rest of the code is being retested now but assumed to be good 😢


https://github.com/open-goal/jak-project/assets/89345505/a9d6cfca-5523-437d-8edc-c74e1f1a13ce

**Comment by Hat-Kid** [comment]
This is not a bug caused by your changes, this happens on `master` as well and appears to be related to the new joint decompression code added in #3369 as using the old decompressor fixes it.

**Comment by xTVaser** [comment]
With https://github.com/open-goal/jak-project/pull/3394 merging, does the bug go away?

**Comment by Zedb0T** [comment]
Ill merge and test in 2/3 hours

**Comment by Zedb0T** [comment]
I can confirm the bug no longer occurs but I am trying to clean one more thing up but dont want to miss the release deadline :( any idea when it will be? 

Not trying to push but I just want this merged, it feels like a diff game with it on its so nice.

**Comment by Zedb0T** [comment]
https://github.com/open-goal/jak-project/assets/89345505/4303b496-2386-4ecb-a424-1c65ac812d02

I want fish and rats game to autostart when talking to these people as you would expect if their cutscenes are skipped then maintain the Play again yes/no menu if you lose/quit out.

I was so certain I had this working but I guess not or I didnt push it, cant see it in the git history anywhere

**Comment by xTVaser** [comment]
> I can confirm the bug no longer occurs but I am trying to clean one more thing up but dont want to miss the release deadline :( any idea when it will be?

It will be in next month's release.  The next release will be in the next day or so and it's best to avoid rush merging something in.

**Comment by xTVaser** [comment]
Re-open if you eventually fix the issue you mentioned here and it's ready to merge https://github.com/open-goal/jak-project/pull/3186#issuecomment-1975548971 It doesn't feel like this is ready or a top priority.

**Pull request closed without merging**